fpdf/text_region.py

```diff
@@ -131,6 +132,96 @@ def build_lines(self, print_sh) -> List[LineWrapper]:
         return text_lines
 
 
+class ImageParagraph:
+    def __init__(
+        self,
+        region,
+        name,
+        align=None,
+        width: float = None,
+        height: float = None,
+        fill_width: bool = False,
+        keep_aspect_ratio=False,
+        top_margin=0,
+        bottom_margin=0,
+        link=None,
+        title=None,
+        alt_text=None,
+    ):
+        self.region = region
+        self.name = name
+        if align:
+            align = Align.coerce(align)
+        self.align = align
+        self.width = width
+        self.height = height
+        self.fill_width = fill_width
+        self.keep_aspect_ratio = keep_aspect_ratio
+        self.top_margin = top_margin
+        self.bottom_margin = bottom_margin
+        self.link = link
+        self.title = title
+        self.alt_text = alt_text
+
+    def build_line(self):
+        # We do double duty as a "text line wrapper" here, since all the necessary
+        # information is already in the ImageParagraph object.
+        self.name, self.img, self.info = self.region.pdf.preload_image(self.name, None)
```

Comment:
Shouldn't `self.img` & `self.info` be initialized in this class constructor?

`None` as a non-named second argument is a bit odd there.
It's not needed, as `dims` already defaults to `None`.
Maybe it should simply be removed there?

Reply:
Accessing those attributes before `build_line()` has been called is an error in either case, and should always throw an error. I think they could even be private, and `render()` could throw an explicit "out-of-sequence" message if they aren't set. Although given that this functionality should only ever be used by one method inherited by a columnar text region, I'm not sure if that's really worth the effort.

I don't mind either way about the None argument. I've never seen any guideline against making default values explicit, though.

Reply:
> Although given that this functionality should only ever be used by one method inherited by a columnar text region, I'm not sure if that's really worth the effort.

Most coding "best practices" are about error prevention, and cases that "should never happen" 😄
But we know from experience that it's often worth being cautious (when it does not take excessive time and/or code).

It's also a matter of readibility.
I usually expect all properties of a `class` to be listed in its constructor.
"Discovering" that a class uses extra properties in one of its method is confusing.

> I don't mind either way about the None argument. I've never seen any guideline against making default values explicit, though.

I'm fine with that if you prefer, but could you please passe it by name? _E.g._ `dims=None`.
It makes code really easier to read.